cloudflare/resource_cloudflare_teams_route.go

```diff
@@ -0,0 +1,134 @@
+	client := meta.(*cloudflare.API)
+
+	tunnelRoute, err := client.GetTunnelRouteForIP(context.Background(), cloudflare.TunnelRoutesForIPParams{
+		AccountID: d.Get("account_id").(string),
+		Network:   d.Get("network").(string),
+	})
+	if err != nil {
+		// FIXME(2022-04-21): Until the API returns a valid v4 compatible envelope, we need to
+		// check if the error message is related to problems unmarshalling the response _or_
+		// an expected not found error.
+		var notFoundError *cloudflare.NotFoundError
+		if strings.Contains(err.Error(), "error unmarshalling the JSON response error body") || errors.As(err, &notFoundError) {
+			d.SetId("")
+			return nil
+		}
+
+		return fmt.Errorf("error reading Tunnel Route for Network %q: %w", d.Id(), err)
+	}
+
+	d.Set("tunnel_id", tunnelRoute.TunnelID)
+	d.Set("network", tunnelRoute.Network)
+
+	if len(tunnelRoute.Comment) > 0 {
+		d.Set("comment", tunnelRoute.Comment)
+	}
+
+	return nil
+}
+
+func resourceCloudflareTeamsRouteCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*cloudflare.API)
+
+	resource := cloudflare.TunnelRoutesCreateParams{
+		AccountID: d.Get("account_id").(string),
+		TunnelID:  d.Get("tunnel_id").(string),
+		Network:   d.Get("network").(string),
+	}
+
+	if comment, ok := d.Get("comment").(string); ok {
+		resource.Comment = comment
+	}
+
+	newTunnelRoute, err := client.CreateTunnelRoute(context.Background(), resource)
+	if err != nil {
+		return fmt.Errorf("error creating Tunnel Route for Network %q: %w", d.Get("network").(string), err)
+	}
+
+	d.SetId(newTunnelRoute.Network)
+
+	return resourceCloudflareTeamsRouteRead(d, meta)
+}
+
+func resourceCloudflareTeamsRouteUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*cloudflare.API)
+
+	resource := cloudflare.TunnelRoutesUpdateParams{
+		AccountID: d.Get("account_id").(string),
+		TunnelID:  d.Get("tunnel_id").(string),
+		Network:   d.Get("network").(string),
+	}
+
+	if comment, ok := d.Get("comment").(string); ok {
+		resource.Comment = comment
+	}
+
+	_, err := client.UpdateTunnelRoute(context.Background(), resource)
+	if err != nil {
+		return fmt.Errorf("error updating Tunnel Route for Network %q: %w", d.Get("network").(string), err)
+	}
+
+	return resourceCloudflareTeamsRouteRead(d, meta)
+}
+
+func resourceCloudflareTeamsRouteDelete(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*cloudflare.API)
+
+	err := client.DeleteTunnelRoute(context.Background(), cloudflare.TunnelRoutesDeleteParams{
+		AccountID: d.Get("account_id").(string),
+		Network:   d.Get("network").(string),
+	})
+	if err != nil {
+		return fmt.Errorf("error deleting Tunnel Route for Network %q: %w", d.Get("network").(string), err)
+	}
+
+	return nil
+}
+
+func resourceCloudflareTeamsRouteImport(d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+	attributes := strings.SplitN(d.Id(), "/", 3)
+
+	if len(attributes) != 2 {
```

Comment:
This import doesn't seem like it'll work as it splits into 3 parts but looks for only 2 here.  Similarly, since the read function only uses the `account_id` and `network` to find the route, shouldn't import just require those two fields?  `tunnel_id` here isn't actually used, as the read call overrides it from the api response.

Reply:
Good catch, I originally implemented it with a different API call where `tunnel_id` was required. I'll send a follow-up PR in a few moments.